Frameworks/CoreGraphics/DWriteWrapper.mm

```diff
@@ -0,0 +1,572 @@
+        // Standardize on uppercase
+        CFStringUppercase(propertyToken, CFLocaleGetSystem());
+
+        // Possible optimization here that can be done by using a dictionary,
+        // but seems excessive given that font names generally don't have more than three modifiers
+        for (const auto& weightMapping : c_weightMap) {
+            if (CFEqual(propertyToken, weightMapping.stringValue)) {
+                *pPropertyContext->weight = weightMapping.dwriteValue;
+                break;
+            }
+        }
+
+        if (CFEqual(propertyToken, CFSTR("UNDEFINED"))) {
+            *pPropertyContext->stretch = DWRITE_FONT_STRETCH_UNDEFINED;
+        } else if (CFEqual(propertyToken, CFSTR("ULTRACONDENSED"))) {
+            *pPropertyContext->stretch = DWRITE_FONT_STRETCH_ULTRA_CONDENSED;
+        } else if (CFEqual(propertyToken, CFSTR("EXTRACONDENSED"))) {
+            *pPropertyContext->stretch = DWRITE_FONT_STRETCH_EXTRA_CONDENSED;
+        } else if (CFEqual(propertyToken, CFSTR("CONDENSED")) || CFEqual(propertyToken, CFSTR("NARROW")) ||
+                   CFEqual(propertyToken, CFSTR("COND"))) {
+            *pPropertyContext->stretch = DWRITE_FONT_STRETCH_CONDENSED;
+        } else if (CFEqual(propertyToken, CFSTR("SEMICONDENSED"))) {
+            *pPropertyContext->stretch = DWRITE_FONT_STRETCH_SEMI_CONDENSED;
+            // skip since this is the default
+            // } else if (CFEqual(propertyToken, CFSTR("NORMAL"))) {
+            //     *pPropertyContext->stretch = DWRITE_FONT_STRETCH_NORMAL;
+
+        } else if (CFEqual(propertyToken, CFSTR("SEMIEXPANDED"))) {
+            *pPropertyContext->stretch = DWRITE_FONT_STRETCH_SEMI_EXPANDED;
+        } else if (CFEqual(propertyToken, CFSTR("EXPANDED"))) {
+            *pPropertyContext->stretch = DWRITE_FONT_STRETCH_EXPANDED;
+        } else if (CFEqual(propertyToken, CFSTR("EXTRAEXPANDED"))) {
+            *pPropertyContext->stretch = DWRITE_FONT_STRETCH_EXTRA_EXPANDED;
+        } else if (CFEqual(propertyToken, CFSTR("ULTRAEXPANDED"))) {
+            *pPropertyContext->stretch = DWRITE_FONT_STRETCH_ULTRA_EXPANDED;
+
+            // skip since this is the default
+            // } else if (CFEqual(propertyToken, CFSTR("NORMAL"))) {
+            //     *pPropertyContext->style = DWRITE_FONT_STYLE_NORMAL;
+        } else if (CFEqual(propertyToken, CFSTR("OBLIQUE"))) {
+            *pPropertyContext->style = DWRITE_FONT_STYLE_OBLIQUE;
+        } else if (CFEqual(propertyToken, CFSTR("ITALIC"))) {
+            *pPropertyContext->style = DWRITE_FONT_STYLE_ITALIC;
+        }
+    };
+
+    CFSetApplyFunction(propertyTokens, initPropertyFromToken, &propertyContext);
+}
+
+/**
+ * Creates a standard DWrite factory object
+ */
+HRESULT _DWriteCreateFactoryInstance(IDWriteFactory** outFactory) {
+    return DWriteCreateFactory(DWRITE_FACTORY_TYPE_SHARED, __uuidof(IDWriteFactory), reinterpret_cast<IUnknown**>(outFactory));
```

Comment:
Brian made this recommendation that we should leave such optimizations to DWrite itself - according to him it's likely that DWrite already caches its factories and by taking manual control over the factory we're messing with its memory management.

---
In reply to: [83500804](https://github.com/Microsoft/WinObjC/pull/1145#discussion_r83500804) [](ancestors = 83500804)